### PR TITLE
RavenDB-19899 Add ButtonGroupWithLabel component

### PR DESCRIPTION
**Pull request by mateuszbartosik**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-19899/Migate-button-group-with-name-to-bs5

### Additional description
Added ButtonGroupWithLabel component; also added story with all button colors

### Type of change
- New feature

### How risky is the change?
- Not relevant

### Backward compatibility
- Not relevant

### Is it platform specific issue?
- No

### Documentation update
- No documentation update is needed 

### Testing by Contributor
- It has been verified by manual testing

### Testing by RavenDB QA team
- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/components/common/ButtonGroupWithLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+interface ButtonGroupWithLabelProps extends ButtonGroupProps {
+    label?: string | JSX.Element;
+}
+
+export function ButtonGroupWithLabel(props: ButtonGroupWithLabelProps) {
+    const { children, label, ...rest } = props;
+
+    return (
+        <ButtonGroup className="flex-vertical btn-group-label" data-label={label} {...rest}>
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/ButtonGroupWithLabel.tsx:12]: Could you also check if it works if you pass icon and span as label? If no, then change in ButtonGroupWithLabelProps from string | JSX.Element to string.

File: src/Raven.Studio/typescript/components/common/ButtonGroupWithLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { ButtonGroup, ButtonGroupProps, Label } from "reactstrap";
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/ButtonGroupWithLabel.tsx]: Are you using Label in this import?

File: src/Raven.Studio/typescript/components/common/Buttons.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+                            </Button>
+                        </ButtonGroupWithLabel>
+                    </div>
+                </>
+            )}
+        </>
+    );
+};
+
+export const GroupWithLabel = boundCopy(Template);
```
Comment (ml054) [inline src/Raven.Studio/typescript/components/common/Buttons.stories.tsx]: If you have 2 different stories to present then no need to use bound copy of template. Just create 2 different stories and export them. Please check other files for reference.

File: src/Raven.Studio/typescript/components/common/ButtonGroupWithLabel.tsx (hunk 1 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/common/ButtonGroupWithLabel.tsx:12]: I believe it's not possible - I've changed it to string.

File: src/Raven.Studio/typescript/components/common/ButtonGroupWithLabel.tsx (hunk 2 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/common/ButtonGroupWithLabel.tsx]: Not anymore - I forgot to remove it. It's gone by now.

File: src/Raven.Studio/typescript/components/common/Buttons.stories.tsx (hunk 3 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/common/Buttons.stories.tsx]: I've split the story into two separate.

**Review by ml054 — APPROVED** [review]

**Review by ml054 — APPROVED** [review]

**Pull request merged**